eldritch/test_events.py

```diff
@@ -752,6 +752,178 @@ def testCombatChoice(self):
 
 # TODO: add tests for going unconscious/insane during a mythos/encounter.
 
+class PurchaseTest(EventTest):
+  def testPurchaseOneAtList(self):
+    buy = Purchase(self.char, "common", 1)
+    self.char.dollars = 3
+    self.assertFalse(buy.is_resolved())
+    self.assertFalse(self.char.possessions)
+    food = items.Food()
+    self.state.common.append(food)
+
+    self.state.event_stack.append(buy)
+    choice = self.resolve_to_choice(CardChoice)
+    self.assertEqual(choice.choices, ["Food for $1", "Nothing"])
+    choice.resolve(self.state, "Food for $1")
+    self.resolve_until_done()
+
+    self.assertTrue(buy.is_resolved())
+    self.assertEqual(self.char.dollars, 2)
+    self.assertEqual(self.char.possessions, [food])
+    self.assertFalse(self.state.common)
+
+  def testPurchaseOneAtListDecline(self):
+    buy = Purchase(self.char, "common", 1)
+    self.char.dollars = 3
+    self.assertFalse(buy.is_resolved())
+    self.assertFalse(self.char.possessions)
+    food = items.Food()
+    self.state.common.append(food)
+
+    self.state.event_stack.append(buy)
+    choice = self.resolve_to_choice(CardChoice)
+    self.assertEqual(choice.choices, ["Food for $1", "Nothing"])
+    choice.resolve(self.state, "Nothing")
+    self.resolve_until_done()
+
+    self.assertTrue(buy.is_resolved())
+    self.assertEqual(self.char.dollars, 3)
+    self.assertFalse(self.char.possessions, [])
+    self.assertEqual(len(self.state.common), 1)
+    self.assertEqual(self.state.common[0], food)
+
+  def testPurchaseOneAtListPoor(self):
```

Comment:
Based on the suspected bug above, please add a test where the first one or two items are too expensive for the player, but the last item is affordable, and then have the player try to purchase the affordable item.

Then assert on the item name in the player's possession.